content/articles/how-to-implement-caching-in-adonisjs-5/index.md

```diff
@@ -0,0 +1,221 @@
+There are more configurations and tweaks for even better Redis usage than the default options we have just accepted. You can read more about Adonis Redis configuration [here](https://docs.adonisjs.com/guides/redis).
+
+### Handling caching in Adonis 5
+Before implementing caching, you should have [cloned the Forum API project](https://github.com/Kaperskyguru/adonisjs-forum-api). You should also have the Redis server, Adonis Redis, and the caching package we will be using in this project installed and configured.
+
+If you have all these checked, we are ready to dive right into the code.
+
+In the world of caching, there are many caching strategies and methods available depending on the structure of your data structure and the different operations performed on the data. You can choose other caching strategies depending on the features and type of the project.
+
+You can read through the [best caching Strategy for Adonis 5](https://masteringbackend.com/posts/caching-strategy-for-restful-api/) which will show you the different caching strategies, and help you choose which one to use depending on the features or the type of project.
+
+We will be using the **cache aside strategy** for all the read operations and a mixture of **write Around** and **write through strategy** for the write operations.
+
+Open your Controller file. 
+
+We already have the `TicketsController` file found in our previous project's `app/Controllers/Http` folder.
+
+First, we will look at the read operations:
+
+```js
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import Forum from 'App/Models/Forum';
+import Cache from '@ioc:Kaperskyguru/Adonis-Cache'
+
+export default class ForumsController {
+
+    // Reading all Forums available
+    public async index({}: HttpContextContract) {
+        const forums = await Cache.remember('forums', 60, async function () {
+
+            // This code is only executed if there was a MISS
+            return await Forum.query().preload('user').preload('posts');
+
+        })
+        return forums
+    }
+
+    // Reading a Single Forum by ID
+    public async update({ auth, request, params}: HttpContextContract){
+        try {
+            const ticket = await Cache.remember('forum_id_' + params.id, 60, async function () {
+            // This code is only executed if there was a MISS
+            return await Forum.find(params.id);
+            })
+
+            if (forum) {
+                forum.title = request.input('title');
+                forum.description = request.input('description');
+                if (await forum.save()) {
+                    await forum.preload('user')
+                    await forum.preload('posts')
+                    return forum
+                }
+            } catch (error) {
+                console.log(error)
+            }
+        }
+    }
+```
+
+The read operations are shown mainly in the `index()` and `show` methods for retrieving all forums and a single forum by ID.
+
```

Comment:
There is no `show()` method in the code block above.

Reply:
DOne